### ISSUE TITLE
Image docker runs as the root user

### ISSUE THREAD

**Issue by DOCPROF** [Division42LLC/division42llc-dotnet-webca#1]
Image division42llc/dotnet-webca runs as the root use.
It is not recommended to use this account.
I tried to modify the docker image but I have this error :

 Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to start Kestrel.
System.AggregateException: One or more errors occurred. (Error -13 EACCES permission denied) ---> Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -13 EACCES permission denied

**Comment by RobSeder** [comment]
I've run across this too in OpenShift - but in that case, we modified OpenShift to allow containers to run as root. I am not sure why, but I do these this shift in thinking in more and more places.

Meanwhile just about every image on Docker Hub runs as root. I don't know how to make an image run with any other account. If you know, please do a pull request. I'm certainly open to making this work in more environments, and I'd like to understand more about this shift away from using root for the container/process 0 approach.

**Comment by DOCPROF** [comment]
I think we should be inspired by S2I
[https://github.com/openshift/source-to-image](https://github.com/openshift/source-to-image)
In the Dockerfile, You must use a type command : USER 1001
That's what I think test.

**Comment by RobSeder** [comment]
When I do that, the container won't start. It errors out with:
```
crit: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to start Kestrel.
System.AggregateException: One or more errors occurred. (Error -13 EACCES permission denied) ---> Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -13 EACCES permission denied
```
An unprivileged user doesn't have permissions to start listening on a port. I don't understand the thinking behind this, and I don't have any ideas on how else to implement it either. The primary process needs root-ish access to be able to run the app. Any ideas?

**Comment by DOCPROF** [comment]
A normal user is not allowed to bind sockets to TCP ports < 1024, you need more permissions than the normal user has. The easiest and quickest solution probably is using sudo to start the process, but that's also the most frowned upon solution. 
Another solution is simply to change the port. I am not an ASP specialist but there has to be a way in Program.cs. To see this article: [https://basicsofwebdevelopment.wordpress.com/2016/09/15/creating-and-hosting-asp-net-core-application-on-linux-nothing-third-party/](https://basicsofwebdevelopment.wordpress.com/2016/09/15/creating-and-hosting-asp-net-core-application-on-linux-nothing-third-party/)

**Comment by RobSeder** [comment]
This has been resolved with commit #7419b3afc23118fb028265e909d90f5748459e45

The web server now listens on port 8080, and the container runs as a non-privileged user. Thanks for the feedback
